core/app/models/spree/product_property.rb

```diff
@@ -8,8 +8,8 @@ class ProductProperty < Spree::Base
     end
 
     validates :property, presence: true
-
     validates :value, db_maximum_length: true
+    validates :property_id, uniqueness: { scope: :product_id }
```

Comment:
@vinay-mittal Please add an option 'allow_nil: true' to skip this validation in case presence fails.